pkg/apis/pipeline/v1/status_validation.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"knative.dev/pkg/apis"
+)
+
+// ValidateEmbeddedStatus checks that the embedded-status feature gate is set to the wantEmbeddedStatus value and,
+// if not, returns an error stating which feature is dependent on the status and what the current status actually is.
+func ValidateEmbeddedStatus(ctx context.Context, featureName, wantEmbeddedStatus string) *apis.FieldError {
```

Comment:
this file shouldn't be needed for Pipeline (only relevant to PipelineRun)

Reply:
I think this is used in the call stack for validateMatrix, which uses the following func:
https://github.com/JeromeJu/pipeline/blob/5e7bc86121ed814b7066b57d5686b5338326abcd/pkg/apis/pipeline/v1/pipeline_types.go#L289

Reply:
Got it that makes sense! In v1, we will only allow this feature flag to be set to "minimal". I think this is OK to stay, but we should also add a similar validation function for PipelineRun (not in this PR) that prevents a v1 PipelineRun from being created unless the feature flag is set to "minimal".

Reply:
Thanks Lee. I've noted this down and will be applying to the PipelineRun change.